### ISSUE TITLE
When running `tauri build`, `bundle_dmg.sh` errors `fails to bundle project`

### ISSUE THREAD

**Issue by joepio** [tauri-apps/tauri#3055]
**Edit**: the `.dmg` files are actually created and function correctly. You can find them in `./target/release/bundle/macos`

### **Describe the bug**
When running `tauri build`, `bundle_dmg.sh` errors `fails to bundle project`.
Looking at the error log below, the issue seems to stem from an Apple script not being authorized: `execution error: Not authorised to send Apple events to Finder. (-1743)`

I tried restarting, upgrading `rustc`, and running `sudo tauri build` - didn't help.

Interestingly, `tauri dev` does work.

### **To Reproduce**
1. `git clone git@github.com:joepio/atomic-data-rust.git`
2. `tauri build`

### **Platform and Versions (required):**
```
Operating System - Mac OS, version 12.0.1 X64

Node.js environment
  Node.js - 16.9.1
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - Not installed

Global packages
  npm - 7.21.1
  yarn - 1.22.17

Rust environment
  rustc - 1.57.0
  cargo - 1.57.0

App directory structure
/target
/server
/cli
/.github
/src-tauri
/lib
/.git
/.vscode

App
  tauri.rs - 1.0.0-beta.8 (no lockfile)
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../server/static
  devPath - ../server/static/tauri-splashscreen.html
package.json not found
```

### **Stack Trace**
```
+ WINX=10
+ WINY=60
+ WINW=500
+ WINH=350
+ ICON_SIZE=128
+ TEXT_SIZE=16
+ FORMAT=UDZO
+ ADD_FILE_SOURCES=()
+ ADD_FILE_TARGETS=()
+ IMAGEKEY=
+ HDIUTIL_VERBOSITY=
+ SANDBOX_SAFE=0
+ SKIP_JENKINS=0
+ MAXIMUM_UNMOUNTING_ATTEMPTS=3
+ POSITION_CLAUSE=
+ HIDING_CLAUSE=
+ [[ - = \- ]]
+ case $1 in
+ VOLUME_NAME='Atomic Server'
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ - = \- ]]
+ case $1 in
+ POSITION_CLAUSE='set position of item "Atomic Server" to {180, 170}
		'
+ shift
+ shift
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ - = \- ]]
+ case $1 in
+ APPLICATION_LINK=480
+ APPLICATION_CLAUSE='set position of item "Applications" to {480, 170}
		'
+ shift
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ - = \- ]]
+ case $1 in
+ WINW=660
+ WINH=400
+ shift
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ - = \- ]]
+ case $1 in
+ HIDING_CLAUSE='set the extension hidden of item "Atomic Server.app" to true
		'
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ - = \- ]]
+ case $1 in
+ VOLUME_ICON_FILE=/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/icon.icns
+ shift
+ shift
+ case $FORMAT in
+ IMAGEKEY='-imagekey zlib-level=9'
+ [[ A = \- ]]
+ [[ -z Atomic Server.app ]]
+++ dirname /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/bundle_dmg.sh
++ cd /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg
++ pwd
+ SCRIPT_DIR=/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg
+ DMG_PATH='Atomic Server_0.28.2_aarch64.dmg'
++ dirname 'Atomic Server_0.28.2_aarch64.dmg'
+ DMG_DIRNAME=.
++ cd .
++ pwd
+ DMG_DIR=/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos
++ basename 'Atomic Server_0.28.2_aarch64.dmg'
+ DMG_NAME='Atomic Server_0.28.2_aarch64.dmg'
+ DMG_TEMP_NAME='/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
++ cd 'Atomic Server.app'
++ pwd
+ SRC_FOLDER='/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/Atomic Server.app'
+ [[ .dmg != \.\d\m\g ]]
+ [[ -z Atomic Server ]]
+ BREW_INSTALL=0
+ AUX_PATH=/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/support
+ '[' 0 -eq 0 ']'
+ test -d /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/support
+ [[ -f /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/Atomic Server.app/.DS_Store ]]
+ echo 'Creating disk image...'
Creating disk image...
+ [[ -f /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg ]]
+ rm -f '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
+ CUSTOM_SIZE=
+ [[ -n '' ]]
+ '[' 0 -eq 0 ']'
+ hdiutil create -srcfolder '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/Atomic Server.app' -volname 'Atomic Server' -fs HFS+ -fsargs '-c c=64,a=16,e=16' -format UDRW '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
...............................................................................................
created: /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg
++ get_size '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
+++ du -s '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
+++ sed -e 's/	.*//g'
++ bytes_size=48224
+++ blocks_to_megabytes 48224
+++ MB_SIZE=25
+++ echo 25
++ echo 25
+ DISK_IMAGE_SIZE=25
+ [[ 0 -eq 1 ]]
+ [[ -n '' ]]
++ expr 25 + 20
+ DISK_IMAGE_SIZE=45
++ hdiutil resize -limits '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
++ awk 'NR=1{print int($1/2048+1)}'
+ MIN_DISK_IMAGE_SIZE=24
+ '[' 24 -gt 45 ']'
+ hdiutil resize -size 45m '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
+ echo 'Mounting disk image...'
Mounting disk image...
+ MOUNT_DIR='/Volumes/Atomic Server'
+ [[ -d /Volumes/Atomic Server ]]
+ echo 'Mounting disk image...'
Mounting disk image...
+ echo 'Mount directory: /Volumes/Atomic Server'
Mount directory: /Volumes/Atomic Server
++ hdiutil attach -readwrite -noverify -noautoopen '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/macos/rw.Atomic Server_0.28.2_aarch64.dmg'
++ egrep --color=never '^/dev/'
++ sed 1q
++ awk '{print $1}'
+ DEV_NAME=/dev/disk4
+ echo 'Device name:     /dev/disk4'
Device name:     /dev/disk4
+ [[ -n '' ]]
+ [[ -n 480 ]]
+ echo 'making link to Applications dir'
making link to Applications dir
+ test -d '/Volumes/Atomic Server/Applications'
+ ln -s /Applications '/Volumes/Atomic Server/Applications'
+ [[ -n '' ]]
+ [[ -n /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/icon.icns ]]
+ echo 'Copying volume icon file '\''/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/icon.icns'\''...'
Copying volume icon file '/Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/icon.icns'...
+ cp /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/icon.icns '/Volumes/Atomic Server/.VolumeIcon.icns'
+ SetFile -c icnC '/Volumes/Atomic Server/.VolumeIcon.icns'
+ [[ -n '' ]]
++ mktemp -t createdmg.tmp.XXXXXXXXXX
+ APPLESCRIPT_FILE=/var/folders/k1/tynj2xbn5nd12wm5qn3sc4600000gn/T/createdmg.tmp.XXXXXXXXXX.ZJcDOTOg
+ [[ 0 -eq 1 ]]
+ [[ 0 -eq 0 ]]
+ applescript_source
+ '[' 0 -eq 0 ']'
+ cat /Users/joep/dev/github/joepio/atomic-data-rust/target/release/bundle/dmg/support/template.applescript
+ sed -e s/WINX/10/g -e s/WINY/60/g -e s/WINW/660/g -e s/WINH/400/g -e s/BACKGROUND_CLAUSE//g -e s/REPOSITION_HIDDEN_FILES_CLAUSE//g -e s/ICON_SIZE/128/g -e s/TEXT_SIZE/16/g
+ perl -pe 's/POSITION_CLAUSE/set position of item "Atomic Server" to {180, 170}
		/g'
+ perl -pe s/QL_CLAUSE//g
+ perl -pe 's/APPLICATION_CLAUSE/set position of item "Applications" to {480, 170}
		/g'
+ perl -pe 's/HIDING_CLAUSE/set the extension hidden of item "Atomic Server.app" to true
		/'
+ sleep 2
+ echo 'Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/k1/tynj2xbn5nd12wm5qn3sc4600000gn/T/createdmg.tmp.XXXXXXXXXX.ZJcDOTOg" "Atomic Server"'
Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/k1/tynj2xbn5nd12wm5qn3sc4600000gn/T/createdmg.tmp.XXXXXXXXXX.ZJcDOTOg" "Atomic Server"
+ /usr/bin/osascript /var/folders/k1/tynj2xbn5nd12wm5qn3sc4600000gn/T/createdmg.tmp.XXXXXXXXXX.ZJcDOTOg 'Atomic Server'
/var/folders/k1/tynj2xbn5nd12wm5qn3sc4600000gn/T/createdmg.tmp.XXXXXXXXXX.ZJcDOTOg:394:406: execution error: Not authorised to send Apple events to Finder. (-1743)
+ echo 'Failed running AppleScript'
Failed running AppleScript
+ hdiutil detach /dev/disk4
"disk4" ejected.
+ exit 64
Error: failed to bundle project

Caused by:
    Shell Scripting Error:`error running bundle_dmg.sh`

 app:spawn Command "/Users/joep/.config/yarn/global/node_modules/@tauri-apps/cli/bin/tauri-cli" failed with exit code: 1 +1ms
```

**Comment by lucasfernog** [comment]
Is this happening when running on CI? If so you need an env variable: #2567

**Comment by joepio** [comment]
Thanks for the quick reply. it's happening locally on my m1 macbook

**Comment by Avineak** [comment]
@lucasfernog 
I am facing similar issues with my m1 mac.
It just tries to open finder  and then fails with same error.


```
info: running bundle_dmg.sh
Error: failed to bundle project

Caused by:
    Shell Scripting Error:`error running bundle_dmg.sh, try running with --verbose to see command output`
   
```

**Comment by probablykasper** [comment]
Running into this as well, can confirm that `CI=true npm run tauri build` works

This should also be fixed when #1731 is done

**Comment by qqpann** [comment]
When setting `CI=true`, it just stops generating the DMG file. Doesn't seem to be a reasonable fix on mac. 🤔

**Comment by probablykasper** [comment]
It works fine for me, so if you're actually not getting a DMG file then that sounds like a separate issue

**Comment by lucasfernog** [comment]
You should get a DMG file, though a not so good looking one :( we'll need to refactor the DMG bundle script to fix this.

**Comment by jamesb93** [comment]
I get this error when trying to run `pnpm tauri build`. Same issue with finder access.

**Comment by lorenzolewis** [comment]
FYI, tracing this back into https://github.com/create-dmg/create-dmg/issues/113.

[Mojave introduced a new plist value](https://developer.apple.com/documentation/bundleresources/information_property_list/nsappleeventsusagedescription) that is required for Apple Script (used in create-dmg) to work with other applications (i.e. Finder).

Maybe we could introduce an OS version check to see if it's Mojave or later and dynamically choose to tack on the CI tag to skip using AppleScript as a workaround until (if) upstream fixes?

**Comment by lucasfernog** [comment]
I wish we could stop using applescript and use something like https://github.com/LinusU/node-appdmg but it's a lot of work to translate that one to Rust.

**Comment by LucienGab** [comment]
when i run CI=true npm run tauri build it still returns the error what am i doing wrong?

**Comment by jswhisperer** [comment]
https://github.com/tauri-apps/tauri/issues/3055#issuecomment-1125221226
same here mac m1

**Comment by jonasmerlin** [comment]
Same here on Mac M1

**Comment by hjmallon** [comment]
There are various threads talking about similar stuff to this. In one of them it recommended to open Photos.app and press 'Get Started' (I haven't used it before on this mac). That actually worked for some reason. 🤷

**Comment by jasonhibbs** [comment]
Commenting because opening Photos.app worked for me on an M1 MacBook Pro…

Before opening Photos.app…

```zsh
Running bundle_dmg.sh
       Error failed to bundle project: error running bundle_dmg.sh: error running bundle_dmg.sh: `failed to run […]/tauri-app/src-tauri/target/release/bundle/dmg/bundle_dmg.sh`
error Command failed with exit code 1.
```

After…

```zsh
Running bundle_dmg.sh
    Finished 2 bundles at:
        […]/tauri-app/src-tauri/target/release/bundle/macos/tauri-app.app
        […]/tauri-app/src-tauri/target/release/bundle/dmg/tauri-app_0.0.0_aarch64.dmg
```

**Comment by kholbekj** [comment]
For me, Photos.app did not accomplish anything on M2 Macbook Air, CI=true option does work!

**Comment by ruanimal** [comment]
It seems is macOS permission related problem, run “tarui build ” manual for the first time, problem solved.

**Comment by rekby** [comment]
I had same problem.
Simple run Photos not help to me.

It was successed after manual run build script (bundle_dmg.sh from targer/release folder) and it build image. Then I unmount own builded image from finder.

**Comment by itsanji** [comment]
I had same problem with `Core i7 Mac` ( `Ventura 13.1` ).
Node version is 16.14.2 and had error with both `yarn` and `npm`.
Error was shown but `.App` file and `.dmg` file was built. 
I tested both and look like it running without any error (yet?)

**Comment by thinksamwp** [comment]
I managed to solve this issue by running pnpm tauri build directly in the macOS terminal.

**Comment by raunakdoesdev** [comment]
The issue for me was that the previously built image was open and bundle_dmg.sh couldn't overwrite it because the old app was running (permission error). Silly problem!

**Comment by emidoots** [comment]
Run `pnpm tauri build --verbose` to get the underlying cause. In my case:

```
hdiutil: create failed - Operation not permitted
could not access /Volumes/Sourcegraph/Sourcegraph.app - Operation not permitted
```

Then forcibly remove the volume:

```
sudo rm -rf /Volumes/Sourcegraph/Sourcegraph.app
```

And that won't fully resolve it (not sure why); you need to restart.

**Comment by maqi1520** [comment]
Same here on Mac M1

But I turned off the updater, it works well
```json
"updater": {
      "active": false,
      "endpoints": ["https://maqi1520.github.io/mdx-editor/install.json"],
      "dialog": true,
      "pubkey": ""
    },
```

**Comment by dukeeagle** [comment]
Tried all the above options on macOS and still receiving the same error! Is this still an issue for anyone else?

**Comment by dukeeagle** [comment]
I fixed it! I think the latest version of macOS is responsible here. 

error when building with yarn in verbose mode `yarn run tauri build -v`:
```
hdiutil: create failed - Operation not permitted
could not access /Volumes/App/App.app - Operation not permitted
```

Fixed by enabling "Full Disk Access" in System Settings for Terminal! (and VSCode for its internal terminal)

**Comment by raunakdoesdev** [comment]
Neat, thanks! Very helpful :))

**Comment by leon-do** [comment]
<img width="827" alt="Screenshot 2023-07-06 at 6 28 37 PM" src="https://github.com/tauri-apps/tauri/assets/19412160/f8c2f5fb-0b73-497a-a6fa-f7f4853694ff">

I had to enable full disk access. Might be a better & safer way.

**Comment by kernelsoe** [comment]
`CI=true pnpm tauri build` works for me! 👍

**Comment by tungtouch** [comment]
i tried it, simple to fix `sudo -E npm run tauri build --verbose` (you will need to use sudo -E in order to expose environment variables.)
make sure disk not full

**Comment by jackiealex** [comment]
see your Mac Finder sidebar, you will see something mounted, just delete it, everyting will be ok

**Comment by JameelKhan9** [comment]
Having the same error, but a different flavour from the looks of the other comments. Just posting my flavour here for anyone that is looking to investigate and wants a clearer picture maybe:

```bash
+ hdiutil create -srcfolder /Users/jameel/Documents/GitHub/tauriapp-desktop-app/src-tauri/target/release/bundle/macos/tauriapp.app -volname tauriapp -fs HFS+ -fsargs '-c c=64,a=16,e=16' -format UDRW /Users/jameel/Documents/GitHub/tauriapp-desktop-app/src-tauri/target/release/bundle/macos/rw.tauriapp_0.0.0_aarch64.dmg
hdiutil: create failed - Read-only file system
       Error [tauri_cli_node] failed to bundle project: error running bundle_dmg.sh
 ELIFECYCLE  Command failed with exit code 1.
```

Seems to be hdiutil permissions related, but using sudo, full disk access, mount permissions didn't help.

**Comment by otwm** [comment]
same error 
```
+ hdiutil create -srcfolder /Users/otwm/WebstormProjects/playground/tauri-app2/src-tauri/target/release/bundle/macos/tauri-app2.app -volname tauri-app2 -fs HFS+ -fsargs '-c c=64,a=16,e=16' -format UDRW /Users/otwm/WebstormProjects/playground/tauri-app2/src-tauri/target/release/bundle/macos/rw.tauri-app2_0.0.0_aarch64.dmg
hdiutil: create failed - 읽기 전용 파일 시스템
       Error [tauri_cli_node] failed to bundle project: error running bundle_dmg.sh
       ```

**Comment by lroolle** [comment]
‌‌In my situation, the `Terminal.app`, which has been granted Full Disk Access, functions correctly. On the other hand, the `Alacritty.app` error in running bundle_dmg.sh

**Comment by musabgultekin** [comment]
These steps solved my issue:
- Give finder access to terminal/vscode (Macos will popup when you first build)
- Remove `src-tauri/target` folder.
- Close and open the terminal/vscode
- Build again

**Comment by kodmanyagha** [comment]
This is about vscode's terminal problem. In mac's terminal there isn't problem.

**Comment by tk-dev** [comment]
This worked for me on macOS Sonoma:

Give Terminal.app the permissions to control Finder.app:
Go to `System-Settings` -> `Privacy & Security` -> `Automation`
There select `Terminal` and below enable the switch `Finder`.

**Comment by theodesp** [comment]
> This worked for me on macOS Sonoma:
> 
> Give Terminal.app the permissions to control Finder.app: Go to `System-Settings` -> `Privacy & Security` -> `Automation` There select `Terminal` and below enable the switch `Finder`.

This solution worked for me as well.

**Comment by nadenf** [comment]
What worked for me is opening Disk Utility and making sure that the image has been properly ejected.

i.e. it should not show up in the sidebar

**Comment by AaronWard** [comment]
```sh
Error failed to bundle project: error running bundle_dmg.sh
```

A mix of the recommended fixes worked for me (for M1 Macbook):

- Make sure settings in automation section have **Finder** enabled:
<img width="703" alt="Screenshot 2024-03-22 at 11 20 10" src="https://github.com/tauri-apps/tauri/assets/22074246/15b2d298-e747-4062-b828-cc699daa3706">

- Deleted `src-tauri/target` folder
- Make sure any other previous/problematic bundles are ejected / deleted or are not currently running
- Close and restart the terminal 
- `npm run tauri build`

**Comment by kristoferfannar-ks** [comment]
I'm having the same error
specs:
* **mac m3 on sonoma 14.5**
* cargo v1.78
* tauri 1.6.0

using `cargo tauri build --target aarch64-apple-darwin --verbose`

```
+ echo 'Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/rm/zcn1js9j6sx82wt_82xm5b9h0000gn/T/createdmg.tmp.XXXXXXXXXX.wHuns9Qx3o" "<project>"'
Running AppleScript to make Finder stuff pretty: /usr/bin/osascript "/var/folders/rm/zcn1js9j6sx82wt_82xm5b9h0000gn/T/createdmg.tmp.XXXXXXXXXX.wHuns9Qx3o" "<project>"
+ /usr/bin/osascript /var/folders/rm/zcn1js9j6sx82wt_82xm5b9h0000gn/T/createdmg.tmp.XXXXXXXXXX.wHuns9Qx3o '<project>'
/var/folders/rm/zcn1js9j6sx82wt_82xm5b9h0000gn/T/createdmg.tmp.XXXXXXXXXX.wHuns9Qx3o:85:89: execution error: Finder got an error: Application isn’t running. (-600)
+ echo 'Failed running AppleScript'
Failed running AppleScript
+ hdiutil detach /dev/disk4
"disk4" ejected.
+ exit 64
       Error [tauri_cli] failed to bundle project: error running bundle_dmg.sh
```

**from a quick search it seems that this is a MacOS error, and the short term solution is to restart the machine, while the long term solution is to report it to apple and wait.**

**Comment by maxktz** [comment]
Same here on Mac M1

**Comment by kyteidev** [comment]
For me it happens randomly. I just build my app again until it works.

**Comment by dondxniel** [comment]
So, at first i was running `cargo tauri build`, but since that failed a couple times and after reading some of the suggestions on how to bye pass the permission problems (some of the suggestions didn't work for me), I just ran `sudo cargo tauri build` and it worked fine.

**Comment by braden-w** [comment]
For me, I needed to grant Full Disk Access to my terminal/IDE (still having this in 2025). Go to System Preferences → Security & Privacy → Privacy tab, select Full Disk Access from the left sidebar, then grant your application in the list.

Rerun the `tauri build` again and grant any further permissions.

Got here by running the script manually; I `cd`ed into `/src-tauri/target/release/bundle/dmg`, then ran

```bash
./bundle_dmg.sh MyApp.dmg ../macos/MyApp.app
```

and saw some error messages.

**Comment by nukeop** [comment]
I'm seeing this in CI here on Github, none of the listed solutions did anything.

**Comment by setoelkahfi** [comment]
MacBook Pro M4, macOS 26.0.1

I was debugging my other issue here https://github.com/tauri-apps/tauri-action/issues/1212#issuecomment-3593795854 and got the same error locally:

```bash
Stapling app...
    Bundling Learn Persian_1.0.0_aarch64.dmg (/Users/setoelkahfi/Repositories/rumilearnpersian/src-tauri/target/release/bundle/dmg/Learn Persian_1.0.0_aarch64.dmg)
     Running bundle_dmg.sh
       Error failed to bundle project error running bundle_dmg.sh: `failed to run /Users/setoelkahfi/Repositories/rumilearnpersian/src-tauri/target/release/bundle/dmg/bundle_dmg.sh`
```

@braden-w solution about Full Disk Access works for me after restarting my terminal. It doesn't solve my original issue though, that is the `Application "App name" cannot be opened. `error.

**Comment by that-one-arab** [comment]
I fixed it by enabling Finder permission in automation for my terminal (Ghostty)

Settings -> Privacy & Security -> Automation

<img width="459" height="235" alt="Image" src="https://github.com/user-attachments/assets/b56336f3-358f-426b-a0e3-2f5af46546b0" />

**Comment by alexfromapex** [comment]
Does anyone know if this is related to Apple certificates? I started having this issue only once I started Notarizing my app, which was shortly after I installed the certificates required for doing that.

Edit: Giving VS Code full disk permissions in system settings fixed it 👍

**Comment by ThatXliner** [comment]
This may be because of Intel mac?

**Comment by eolix** [comment]
> I fixed it by enabling Finder permission in automation for my terminal (Ghostty)
> 
> Settings -> Privacy & Security -> Automation
> <img alt="Image" width="459" height="235" src="https://private-user-images.githubusercontent.com/91668821/535179313-b56336f3-358f-426b-a0e3-2f5af46546b0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzkwNTg3MjAsIm5iZiI6MTc3OTA1ODQyMCwicGF0aCI6Ii85MTY2ODgyMS81MzUxNzkzMTMtYjU2MzM2ZjMtMzU4Zi00MjZiLWEwZTMtMmY1YWY0NjU0NmIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA1MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNTE3VDIyNTM0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwMjZiOGE3YTJmMTNmM2YzZGQ1ZTBhNWM1OTdkMjUyZGViM2U0YmVhZTllNDExNmI1YzJjOWRkZWJjOTNiYmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.FsPO4E5jJ2vi8vDxNzCBHWHh0Xq2afSEAdGymCWsHW8">

This is proof that we can't get rid of human comments / stackoverflow. LLM spent hours modifying Rust code to try and fix this. What a world we live in.
